bindings/python/dlite-entity-python.i

```diff
@@ -507,17 +512,23 @@ def get_instance(id: "str", metaid: "str"=None, check_storages: "bool"=True) ->
             return object.__getattribute__(self, name)
         d = object.__getattribute__(self, '__dict__')
         if name in d:
-            return d[name]
+            value = d[name]
         elif self.has_property(name):
-            return _get_property(self, name)
+            value = _get_property(self, name)
         elif self.has_dimension(name):
-            return self.get_dimension_size(name)
+            value = self.get_dimension_size(name)
         else:
             raise AttributeError('Instance object has no attribute %r' % name)
+        if isinstance(value, np.ndarray) and self.is_frozen():
+            value.flags.writeable = False  # ensure immutability
+        return value
 
     def __setattr__(self, name, value):
         if name == 'this':
             object.__setattr__(self, name, value)
+        elif self.is_frozen():
+            raise DLiteError(
+                'frozen instance does not support attribute assignment')
```

Comment:
Shouldn't this be the first check?

Reply:
Hmm. We should add another issue about avoiding using proper keywords as special keywords, so users don't run into an issue with these special keywords suddenly.

I suggest to change the special keywords to something more straight-forward. E.g., `uuid` -> `_dlite_uuid` or something?